sdk/nodejs/tests/tests.ts

```diff
@@ -0,0 +1,550 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as bcryptjs from "bcryptjs";
+import { assert } from "chai";
+import * as os from "os";
+import * as sut from "../index";
+
+const validator = require("validator");
+
+describe("abs", () => {
+    it("works", () => {
+        assert.equal(sut.abs(1), 1);
+        assert.equal(sut.abs(-1), 1);
+        assert.equal(sut.abs(-3.14), 3.14);
+        assert.equal(sut.abs(42.001), 42.001);
+    });
+});
```

Comment:
Are these tests hooked up to a test harness?  Would be nice to have an `npm` script to run them - and to hook that into `make`.

Reply:
They are not currently, but will be prior to this PR being ready to merge.